widgetssdk/src/main/java/com/glia/widgets/core/queue/QueueRepository.kt

```diff
@@ -32,57 +40,65 @@ internal class QueueRepositoryImpl(private val gliaCore: GliaCore, private val c
         .doOnSubscribe { fetchQueues() }
         .distinctUntilChanged()
 
-    override val relevantQueueIds: List<String>
-        get() = _queuesState.value
-            ?.let { it as? QueuesState.Queues }
-            ?.run { queues.map { it.id } }
-            .orEmpty()
+    private val _relevantQueueIds: Flowable<List<String>>
+        get() = queuesState
+            .filter { it !is QueuesState.Loading }
+            .map { it.queuesOrEmpty() }
+            .map { getIds(it) }
 
-    init {
-        subscribeToQueues()
-        subscribeToQueueUpdates()
+    override val relevantQueueIds: Single<List<String>>
+        get() = _relevantQueueIds.firstOrError()
+
+    override fun initialize() {
+        fetchQueues()
     }
 
     private fun fetchQueues() {
-        if (siteQueues.value?.isSuccess != true) {
+        if (siteQueues.value?.isSuccess != true && gliaCore.isInitialized) {
```

Comment:
I think it's logical to change the order to `(gliaCore.isInitialized && siteQueues.value?.isSuccess != true)`.